docs/jumpstart/week-6.md

```diff
@@ -0,0 +1,711 @@
+}
+```
+
+To run these tests once they've been added, you can just run `mvn` or `mvn install` if you want to do it through Maven.
+The tests will be found and run automatically for you, since Maven is set up to look for IntelliJ tests.
+If you'd prefer to do it in IntelliJ, all you have to do is hit the green play button, and choose Run or Debug.
+
+>Note: for Maven to figure out that the method is a test, the method name needs to have the word 'test' somewhere in it.
+
+#### BeforeEach, BeforeAll, AfterEach, and AfterAll
+
+JUnit provides `@BeforeEach`, `@BeforeAll`, `@AfterEach`, and `@AfterAll` annotations. These allow you to perform 
+methods before/after every/each test if it helps to set up or clean up tests without you having to call them manually.
+
+>Note: for the "All" methods, the methods have to be static to work.
+
+Here is an example:
+```java
+@BeforeEach
+public void setup() {
+    this.db = new DatabaseImplementation();
+    this.processor = new ProcessorThing(this.db);
+}
+```
+
+The above code will then run before each test in the test suite and set up the processor and database for you.
+
+#### Parameterized Tests
+
+You can also run multiple tests using different input parameters by using the `@ParameterizedTest` 
+annotation. 
+
+You can add specific inputs to the `@ParameterizedTest` by adding one or more 'source' annotations.
+Three of the most basic ones are `@NullSource`, `@EmptySource` (provides an empty String, List, Set, Map, or any 
+array type), and the combination `@NullAndEmptySource`. Those can be combined with `@ValueSource` (or you can use them 
+alone if that suits your needs better) to test with a simple set of values.
+
+The `@ValueSource` annotation allows you to provide a list of values of any atomic type (short, byte, int, long, float,
+double, char, boolean), as well as String and `Class` types. 
+
+>Note: the input to the annotation must be the pluralized name of the type you're inputting. As an example, a list of 
+>integer inputs would be `@ValueSource(ints = {1, 2, 3, ...})`, and booleans would 
+>be `@ValueSource(booleans = {true, false})`.
+
+The `@EnumSource` conveniently provides all sub-types of a provided enum class.
+
+If none of those suit your needs, you can also use `@MethodSource`, `@CsvSource`, `@CsvFileSource`, and 
+`@ArgumentSource`.
+
+You can see all of the sources of parameterized arguments [on the JUnit 
+docs](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources).
+
+Here is an example of a parameterized test using integer inputs:
+
+```java
+@ParameterizedTest
+@ValueSource(ints = {1, 2, 3})
+@NullSource
+// Notice how we're using the wrapped integer class "Integer" since we're inputting a null source.
+public void testParameterizedTest(Integer value) { 
+    if (value == null) {
+        assertNull(value);
+    }
+    else {
+        assertTrue(value > 0 && value < 4);
+    }
+}
+```
+
+##### Externals Class
```

Comment:
Shouldn't this section be introduced after Mockito?

Reply:
I moved it into the mockito section right before the example